app/controllers/find/search/subjects_controller.rb

```diff
@@ -18,6 +18,8 @@ def create
             has_vacancies: default_vacancies,
           ))
         else
+          @subjects_form.errors.clear
+          @subjects_form.primary? ? @subjects_form.errors.add(:subjects, :primary_subject) : @subjects_form.errors.add(:subjects, :secondary_subject)
```

Comment:
This looks like it should belong to the form

Reply:
Yeah, I gave that a whirl, but the actual form class only has access to params once it is instantiated. I think we would require 2 forms, a primary and secondary etc

Reply:
```
module Find
  class SubjectsForm

    # remove
    # validates :subjects, presence: true

    validate :subjects_has_been_selected

    def subjects_has_been_selected
      if subjects.blank? && age_group.present?
        @subjects_form.errors.add(:subjects, "#{age_group}_subject")
      end
    end
   # brevity
  end
end
```

You just need to move your current logic down to the form, as you have already have the context on it.

Reply:
Nice